atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -1969,16 +1969,12 @@ private void commitWrites(TransactionService transactionService) {
             // This must happen before conflict checking, otherwise we could complete the checks and then have someone
             // else write underneath us before we proceed (thus missing a write/write conflict).
             // Timing still useful to distinguish bad lock percentiles from user-generated lock requests.
-            LockToken commitLocksToken = timedAndTraced("commitAcquireLocks", this::acquireLocksForCommit);
-            closer.register(() -> {
-                // Not timed because tryUnlock() is an asynchronous operation.
-                traced("postCommitUnlock", () -> timelockService.tryUnlock(ImmutableSet.of(commitLocksToken)));
-            });
+            LockToken commitLockToken = transactionLocksManager.registerLock(
```

Comment:
Just a cleanup, we now register locks with the manager, and manager registers itself with the closer. The only thing to think about is whether there are any correctness worries with the order of releasing locks wrt. any other cleanup we do in a transaction.

And there should not be any: nominally, we unlock the locks asynchronously from this call ANYWAY, which indicates that we don't care WHEN these happen.